XenAdmin/Wizards/HAWizard.cs

```diff
@@ -108,7 +104,32 @@ protected override void UpdateWizardContent(XenTabPage senderPage)
             {
                 xenTabPageHaFinish.HeartbeatSrName = xenTabPageChooseSR.SelectedHeartbeatSR.Name;
                 xenTabPageHaFinish.Ntol = xenTabPageAssignPriorities.Ntol;
-                xenTabPageHaFinish.RestartPriorities = xenTabPageAssignPriorities.getCurrentSettings().Values;
+
+                int alwaysRestartHighPriority = 0, alwaysRestart = 0, bestEffort = 0, doNotRestart = 0;
+                foreach (VM.HA_Restart_Priority priority in xenTabPageAssignPriorities.getCurrentSettings().Values)

```

Comment:
Had another look at the wizard workflow: UpdateWizardContent is indeed called in PageLeave, so it seems to me the alternative way to do this is calculate and set the field each time the settings change on the page, so when we leave the page we'll only have to get that value, without any calculation at all.